src/rust_helpers/weak_self.rs

```diff
@@ -0,0 +1,79 @@
+// copied from https://github.com/eun-ice/weak-self
+
+use std::cell::UnsafeCell;
+use std::fmt;
+use std::rc::{Rc, Weak};
+
+pub struct WeakSelf<T: ?Sized> {
+    cell: UnsafeCell<Option<Weak<T>>>,
+}
+
+impl<T: ?Sized> Clone for WeakSelf<T> {
```

Comment:
And I implemented `Clone` for `WeakSelf`

Reply:
What does the "?" in front of "Sized" do?

Reply:
I didn't write this code (copied from existing `weak_self` crate) but I think that by default generics have a `Sized` bound and so `?Sized` is the syntax for "opting out" of that